src/pages/requests/index.tsx

```diff
@@ -12,50 +16,56 @@ import { Sidebar } from "../../others/components/Sidebar";
 import { MultiTab } from "../../others/components/MultiTab";
 import { CollapsibleTable } from "../../others/components/CollapsibleList";
 import { layerStyle } from "../../others/components/map/CircleLayerStyle";
-import { mapAidRequestsToFeatures } from "../../others/helpers/map-utils";
-import { processAidRequests } from "../../others/helpers/process-aid-request";
-import { useSidebarContext } from "../../others/components/sidebar-context";
-import { FilterItem, useFilter } from "../../others/contexts/filter";
-import { DecodedLocation, DecodedAidRequest } from "../../others/helpers/decode-aid-request";
-import { mapLocationsToTableData, mapCategoriesToTableData } from "./map-to-table-data";
+import { mapAidRequestsToFeatures, adaptToMap } from "../../others/helpers/map-utils";
+import {
+  sortDates,
+  filterByCategoryIds,
+  groupByCities,
+  groupByCategories,
+  groupedByCitiesToTableData,
+  groupedByCategoriesToTableData,
+  translateToLocation,
+  translateToSupply,
+} from "../../others/helpers/aid-request-helpers";
 
 export function Requests() {
   const { t } = useTranslation();
-  const { data: cities } = useLocationsQuery();
-  const { data: supplies } = useSuppliesQuery();
   const { data: aidRequests } = useAidRequestQuery();
+  const { locations, supplies, translateLocation, translateSupply } = useDictionaryContext();
+
   const filterContext = useFilter();
 
   const addFilter = filterContext.addFilter;
 
-  const { decodedAndGroupedByLocation, decodedAndGroupedByCategory } = useMemo(() => {
-    return processAidRequests(cities, supplies, aidRequests);
-  }, [cities, supplies, aidRequests]);
+  // First create a lookup table for all aid requests grouped by dates and memoise it
+  // TODO: consider discarding dates from longer than 7 days ago
```

Comment:
I'm not sure we need this comment. Backend API will have a filtering for the necessary dates, so frontend is just responsible to handle what it is given.

Reply:
Didn't know that, thanks for explanation!